Tests/UnitTests/Purchasing/Purchases/PurchasesDelegateTests.swift

```diff
@@ -76,4 +76,38 @@ class PurchasesDelegateTests: BasePurchasesTests {
         expect(self.notificationCenter.observers).to(haveCount(2))
     }
 
+    // See https://github.com/RevenueCat/purchases-ios/issues/2410
+    func testDelegateWithGetCustomerInfoCallDoesNotDeadlock() throws {
+        final class GetCustomerInfoPurchasesDelegate: NSObject, PurchasesDelegate {
+            func purchases(_ purchases: Purchases, receivedUpdated customerInfo: CustomerInfo) {
```

Comment:
Should we add a check to make sure this is actually called? To make sure nothing changes that could cause us to miss the important part of the test (this might be part of another test so it might be ok).

Reply:
Yeah that's part of another test but it doesn't hurt.